exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/SpanAdapter.java

```diff
@@ -104,15 +135,17 @@ static Span toProtoSpan(SpanData spanData) {
     builder.setDroppedAttributesCount(
         spanData.getTotalAttributeCount() - spanData.getAttributes().size());
     for (EventData event : spanData.getEvents()) {
-      builder.addEvents(toProtoSpanEvent(event));
+      builder.addEvents(toProtoSpanEvent(event, threadLocalCache));
     }
     builder.setDroppedEventsCount(spanData.getTotalRecordedEvents() - spanData.getEvents().size());
     for (LinkData link : spanData.getLinks()) {
-      builder.addLinks(toProtoSpanLink(link));
+      builder.addLinks(toProtoSpanLink(link, threadLocalCache));
     }
     builder.setDroppedLinksCount(spanData.getTotalRecordedLinks() - spanData.getLinks().size());
     builder.setStatus(toStatusProto(spanData.getStatus()));
-    return builder.build();
+    Span span = builder.build();
+    builder.clear();
```

Comment:
this line of code is incredibly important. I think it deserves a comment. :)

Reply:
Yup added comments. Also confirmed any missing clear results in failing tests.